### DISCUSSION TITLE
The detected CUDA version (12.2) mismatches the version that was used to compile PyTorch(11.7)

### DISCUSSION THREAD

**Discussion by kathy060415** [facebookresearch/detectron2#5071] [Q&A]
<img width="652" alt="image" src="https://github.com/facebookresearch/detectron2/assets/72689388/799ac248-1162-4276-a500-9e341bc76b05">

I am working on an external GPU using WinSCP. I have downloaded torch using 'conda install pytorch torchvision torchaudio pytorch-cuda=11.7 -c pytorch -c nvidia' through putty.exe.

I have checked the cuda version using torch.version.cuda and it shows up as 11.7.

However, when I try to run !python -m pip install 'git+https://github.com/facebookresearch/detectron2.git' on Jupyter Notebook, I get the error saying that 'The detected CUDA version (12.2) mismatches the version that was used to compile PyTorch (11.7). Please make sure to use the same CUDA versions.'.

This is what is running in my .sh file on WinSCP to run jupyter notebook on external GPU.
<img width="293" alt="Screenshot 2023-08-28 211441" src="https://github.com/facebookresearch/detectron2/assets/72689388/499f5a64-764a-4b34-a1e6-124fcbba8b3e">

My conda environment has the following cuda packages
<img width="437" alt="image (1)" src="https://github.com/facebookresearch/detectron2/assets/72689388/38503846-8470-4690-a8f8-e34af6cce027">

The external GPU I am working with comes with the following shared applications:
<img width="391" alt="Screenshot 2023-08-28 211125" src="https://github.com/facebookresearch/detectron2/assets/72689388/eae6ffb0-21d5-4eb4-89e7-cdbdb7d5baf5">

I am wondering when pytorch goes around it detects another CUDA? I have been erasing and making several conda environments to solve this problem, but I am coming up with the same error.

Can someone please suggest a solution to this? Would there be a way to specify a cuda version when running a command? Thanks.

**Comment by bmaxdk** [comment]
There could be instances where it detects a system-wide CUDA installation that is different from the one in your conda environment.
